tools/generate_std_names.py

```diff
@@ -62,32 +77,72 @@ def process_name_table(tree, element_name, *child_elements):
     """
     for elem in tree.iterfind(element_name):
         sub_section = {}
-
         for child_elem in child_elements:
-            found_elem = elem.find(child_elem)
-            sub_section[child_elem] = found_elem.text if found_elem is not None else None
-
+            sub_section[child_elem] = found_or_none(elem.find(child_elem))
         yield {elem.get("id") : sub_section}
 
 
-def to_dict(infile, outfile):
-    values = {}
-    aliases = {}
-
+def prettydict(outfile, varname, data):
+    """Pretty formatted output of the data (dict) assigned to the variable 'varname'."""
+    outfile.write(f'{varname} = {{\n')
+    for k, v in dict(sorted(data.items())).items():
+        outfile.write(f'    "{k}": "{v}",\n')
+    outfile.write("}\n\n")
+
+
+def decode_version(outfile, tree):
+    """Decode the version information in the xml header information."""
+    version = {}
+    for elem in ["table_name", "version_number", "last_modified", "institution", "contact"]:
+        version[elem] = found_or_none(tree.find(elem))
+    if version["table_name"] is None:
+        if (version["institution"] == "Centre for Environmental Data Analysis"
+                and version["contact"] == "support@ceda.ac.uk"):
+            version["table_name"] = "CF-StdNameTable"
+        else:
+            version["table_name"] = "USER-StdNameTable"
+    prettydict(outfile, "VERSION", version)
+    version_string = "-".join(version[k] for k in ["table_name", "version_number"])
+    outfile.write(f'CONVENTIONS_STRING = "{version_string}"\n\n')
```

Comment:
Note that this was sorted in #5423, albeit in a different way.

Presumably no longer necessary in this PR?